finat/argyris.py

```diff
@@ -37,108 +37,58 @@ def basis_transformation(self, coordinate_mapping):
             V[s, s] = Literal(1)
             for i in range(2):
                 for j in range(2):
-                    V[s+1+i, s+1+j] = Indexed(J, (j, i))
-            V[s+3, s+3] = Power(Indexed(J, (0, 0)), Literal(2))
-            V[s+3, s+4] = Product(Literal(2),
-                                  Product(Indexed(J, (0, 0)),
-                                          Indexed(J, (1, 0))))
-            V[s+3, s+5] = Power(Indexed(J, (1, 0)), Literal(2))
-            V[s+4, s+3] = Product(Indexed(J, (0, 0)),
-                                  Indexed(J, (0, 1)))
-            V[s+4, s+4] = Sum(Product(Indexed(J, (0, 0)),
-                                      Indexed(J, (1, 1))),
-                              Product(Indexed(J, (1, 0)),
-                                      Indexed(J, (0, 1))))
-            V[s+4, s+5] = Product(Indexed(J, (1, 0)),
-                                  Indexed(J, (1, 1)))
-            V[s+5, s+3] = Power(Indexed(J, (0, 1)), Literal(2))
-            V[s+5, s+4] = Product(Literal(2),
-                                  Product(Indexed(J, (0, 1)),
-                                          Indexed(J, (1, 1))))
-            V[s+5, s+5] = Power(Indexed(J, (1, 1)), Literal(2))
+                    V[s+1+i, s+1+j] = J[j, i]
+            V[s+3, s+3] = Power(J[0, 0], Literal(2))
```

Comment:
All these Power nodes are just squares of short expressions, perhaps nicer to just write them as a product, for example:
```python
V[s+3, s+3] = J[0, 0]*J[0, 0]
```